ckanext/opendatani/helpers.py

```diff
@@ -179,8 +179,9 @@ def prepare_csv_report(resource):
     :return: a string containing the csv_id of the created archive
     :rtype: string
     """
-    if toolkit.c.user is None:
-        return None
+    if not toolkit.c.user or not resource:
+        log.error('Resource or user not found.')
+        return
 
     file_name = uuid.uuid4().hex + '.csv'
     file_path = '/var/lib/ckan/storage/tmp/' + file_name
```

Comment:
You can save directly into `/tmp/` is there a reason it's saved in `/var/lib/ckan/storage`? And if there is a reason that should not be hardcoded. You should get it from `config.get('ckan.storage_path')`

Reply:
It seems that `storage` gets mounted so it's accessible like this:

`http://ckan-dev:5000/report/test_org_1/tmp/fd8ea6b9352f4d5fb9b758847264da02.csv`

And when stored in just `/tmp/`, not as easily accessible.

It was the cleanest solution I found. I'll change to `ckan.storage_path`

Reply:
What if you remove `/tmp/` in that case?

Reply:
And why URL aslo has to include `tmp`?

Reply:
I tried storing in `/tmp/` instead of `/storage/tmp/` and in that case, it redirects to `http://ckan-dev:5000/report/test_org_1/<csv_id>` properly, but results in `forbidden` on ckan_admin and my org admin account...

Reply:
What if you store ins `storage/` but remove `tmp`?

Reply:
I'm getting the same error there as well. I'll look a bit more. Also, to correct what I said, it redirects to `http://ckan-dev:5000/tmp/<csv_id>` not `http://ckan-dev:5000/report/test_org_1/tmp/<csv_id>`

Reply:
Okay. So the issue is that in the html file, I was removing '/tmp/', but I needed to leave the `/`. One thing is that I'll need to remove these files myself now, right? Whereas, the `tmp` folder gets cleared automatically (after a reboot, I think?).

Reply:
I decided to overwrite the old files every time the page loads, to avoid having to delete old files. I wasn't able to implement a good solution as far as deleting the files. This is because the files only get created when the page loads, and calls I tried to make to a `remove_reports` function were called on loading of the page (deleting the files before the button is clicked). I even tried a small JS script that checked when the user left the page before deleting the files, but everything still seemed to load as soon as I navigated to the page, deleting the files before I could download them. 

Anyway, there will only ever be two files (CSV and JSON) for each org at a given time now. I don't think it makes much since to store a backlog on the server.

Reply:
I also changed the file naming convention to `publisher-report-<org name>-<date>.csv`.

Reply:
Great, think this is fine for now